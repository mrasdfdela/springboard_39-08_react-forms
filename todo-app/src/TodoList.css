div {
  margin: 20px;
}
ul {
  list-style-type: none;
  padding-left: 0;
}