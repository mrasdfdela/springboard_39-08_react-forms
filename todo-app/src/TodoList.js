import React, {useState} from 'react';
import { v4 as uuid } from 'uuid';

import Todo from './Todo';
import NewTodoForm from './NewTodoForm';

import './TodoList.css';

function TodoList() {
  const [ todos, setTodos ] = useState([
    {'task':'take out trash', id: uuid()}
  ]);

  const addTodo = (newTodo) =>{
    setTodos( todos => [ ...todos, {...newTodo, id:uuid() }] );
  }

  const deleteTodo = (id)=> {
    setTodos( todos => todos.filter( todo => todo.id !== id) );
  }
  
  return (
    <div>
      <NewTodoForm addTodo={addTodo} />
      <ul className="TodoList"> 
        {todos.map((todo) => (
          <Todo 
            key={todo.id}
            deleteTodo={ ()=> deleteTodo(todo.id) }
            task={todo.task} 
            id={todo.id} 
          />
        ))}
      </ul>
    </div>
  );
}

export default TodoList;