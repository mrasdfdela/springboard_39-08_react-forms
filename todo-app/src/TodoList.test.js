import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import TodoList from './TodoList';

it('renders without crashing', () => {
  render(<TodoList />);
});

it('matches snapshot', () => {
  const {asFragment} = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it('adds and deletes todos correctly', ()=>{
  const { queryByText, queryByLabelText } = render(<TodoList />);
  // adds and checks new to-do
  const formTodo = queryByLabelText("Add To Do's!");
  const addBtn = queryByText('Submit');
  fireEvent.change(formTodo, {target: {value: "take out the trash"}} );
  fireEvent.click(addBtn)
  
  expect(queryByText("take out the trash")).toBeInTheDocument();
  
  // deletes and checks new to-do has been removed
  const deleteBtn = queryByText('X');
  
  fireEvent.click(deleteBtn);
  expect(queryByText("X")).not.toBeInTheDocument();
  expect(queryByText("take out the trash")).not.toBeInTheDocument();
});