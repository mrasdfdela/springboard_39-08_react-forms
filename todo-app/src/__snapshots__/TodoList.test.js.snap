// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="task"
      >
        Add To Do's!
      </label>
      <br />
      <input
        id="task"
        name="task"
        placeholder="Add todo here"
        type="text"
        value=""
      />
      <button>
        Submit
      </button>
    </form>
    <ul
      class="TodoList"
    />
  </div>
</DocumentFragment>
`;
